locust/argument_parser.py

```diff
@@ -351,11 +351,9 @@ def setup_parser_arguments(parser):
     )
     web_ui_group.add_argument(
         "--web-auth",
-        type=str,
-        dest="web_auth",
-        metavar="<username:password>",
-        default=None,
-        help="DEPRECATED. See https://github.com/locustio/locust/issues/2517 Turn on Basic Auth for the web interface. Should be supplied in the following format: username:password ",
+        default=False,
```

Comment:
I think we should use a new argument name to avoid confusion. if someone says `locust --web-auth ...` there needs to be a clear message about what has happened (and maybe a link to the docs). Maybe the new name should be `use_auth`?

Reply:
Cool! But can you leave `--web-auth` still in there? 

Like with for example --expect-slaves, using `help=configargparse.SUPPRESS,` and exiting if someone specifies it https://github.com/locustio/locust/blob/648781401dac8b3112f9e818f5ce7cf0903e55c6/locust/main.py#L146

Reply:
Yup makes sense :+1: 
![image](https://github.com/locustio/locust/assets/59211596/36bf9f08-d4c9-405a-b950-ff369e5ec702)

Reply:
btw we can't use `--use-auth` because of this
![image](https://github.com/locustio/locust/assets/59211596/f98c42ab-b5c9-4ae9-a4f9-55e122017c80)